permissions/middleware.go

```diff
@@ -64,13 +70,53 @@ func AuthenticationMiddleware() gin.HandlerFunc {
 				}
 
 				c.Set("ID_TOKEN_CLAIMS", claims)
+
+				mdb := c.MustGet("MDB").(*sql.DB)
+				user, err := getOrCreateUser(mdb, &claims)
+				if err != nil {
+					c.AbortWithError(http.StatusBadRequest, err).SetType(gin.ErrorTypePrivate)
+					return
+				}
+
+				c.Set("USER", user)
 			}
 		}
 
 		c.Next()
 	}
 }
 
+func getOrCreateUser(mdb *sql.DB, claims *IDTokenClaims) (*models.User, error) {
+	user, err := fetchUserFromDB(mdb, claims.Sub)
+	if err != nil || user != nil {
+		return user, err
+	}
+
+	user = &models.User{
+		AccountID: null.StringFrom(claims.Sub),
+		Email:     claims.Email,
+		Name:      null.StringFrom(fmt.Sprintf("% %", claims.GivenName, claims.FamilyName)),
+	}
+
+	tx, err := mdb.Begin()
+	utils.Must(err)
+
+	if err := user.Insert(tx); err != nil {
+		utils.Must(tx.Rollback())
+	} else {
+		utils.Must(tx.Commit())
+	}
+	return fetchUserFromDB(mdb, claims.Sub)
```

Comment:
Not sure you have to fetch it from DB here. Consider simply returning `user`.
If you must fetch then there is a simple method on User. `Refresh` or `Reload` can't recall the actual name